From 432cd63e585ffc9a9f9268f33c8c4a1eb95f145a Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 13 Jun 2013 11:38:01 +0300
Subject: [PORT FROM MAIN] Telephony: configure retry alarm only if data is allowed

BZ: 106926

Currently, reconnect alarm is getting configured even
when the data is not allowed. Reconnect alarm is 5seconds.
If the user enables the data again just when the alarm goes
off, SETUP_DATA_CALL will get triggered with a Data connection
channel which will get reset to null on retry intent handling.
Even if the SETUP_DATA_CALL succeeds, framework considers
this as an due to no data channel associated with the apnContext.
So, data will be deactivated and won't be retried again.

Configure the retry alarm only if the data is enabld and
allowed.

Change-Id: Ib8f0e18eb3eabc187af41cbb2e568b9823e93395
Orig-Change-Id: Ic986ee2376d04a15d360ca5a3fb8381378f07db4
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index 5d598df..fdec736 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -1676,8 +1676,8 @@ public final class DcTracker extends DcTrackerBase {
             }
         }
 
-        // If APN is still enabled, try to bring it back up automatically
-        if (apnContext.isReady() && retryAfterDisconnected(apnContext.getReason())) {
+        // If data is still enabled and allowed, try to bring it back up automaticall
+        if (isDataAllowed(apnContext) && retryAfterDisconnected(apnContext.getReason())) {
             SystemProperties.set(PUPPET_MASTER_RADIO_STRESS_TEST, "false");
             // Wait a bit before trying the next APN, so that
             // we're not tying up the RIL command channel.
-- 
1.7.4.1

